File: src/Entity/ServiceV.php
<?php

namespace App\Entity;

use App\Repository\ServiceVRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ServiceVRepository::class)]
class ServiceV
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "le champs est vide")]
    private ?float $prix = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "le champs est vide")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "le champs est vide")]
    private ?string $description = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $date = null;

    #[ORM\ManyToOne(inversedBy: 'serviceVs')]
    #[ORM\JoinColumn(nullable: false)]
    private ?CategorieV $category = null;

    #[ORM\Column]
    private ?int $viewCount = 0;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getCategory(): ?CategorieV
    {
        return $this->category;
    }

    public function setCategory(?CategorieV $category): self
    {
        $this->category = $category;

        return $this;
    }

    public function getViewCount(): ?int
    {
        return $this->viewCount;
    }

    public function setViewCount(int $viewCount): self
    {
        $this->viewCount = $viewCount;

        return $this;
    }
}

File: CodeNameOne-Formation-Youtube-main/src/com/mycomany/entities/ServiceV.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycomany.entities;

/**
 *
 * @author Mynet
 */
public class ServiceV {
 
    private int id;
    private double prix;
    private String nom;
    private String description;
    private String date;
    private CategorieV category;
    private int viewCount;
  
 public ServiceV(){      
}
 public ServiceV(int id,double prix,String nom,String description,String date,CategorieV category,int viewCount){ 
 
     this.id=id;
     this.prix=prix;
     this.nom=nom;
     this.description=description;
     this.date=date;
     this.category=category;  
     this.viewCount=viewCount;
}
public ServiceV(double prix,String nom,String description,String date,CategorieV category,int viewCount){ 
 
     this.prix=prix;
     this.nom=nom;
     this.description=description;
     this.date=date;
     this.category=category;  
     this.viewCount=viewCount;
}
public ServiceV(double prix,String nom,String description,String date,CategorieV category){ 
 
     this.prix=prix;
     this.nom=nom;
     this.description=description;
     this.date=date;
     this.category=category;  

}

    public int getViewCount() {
        return viewCount;
    }

    public double getPrix() {
        return prix;
    }

    public String getNom() {
        return nom;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    public CategorieV getCategory() {
        return category;
    }

    public int getId() {
        return id;
    }

    public void setViewCount(int viewCount) {
        this.viewCount = viewCount;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setCategory(CategorieV category) {
        this.category = category;
    }

}
